src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,19 +46,31 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        String[] addressParts = new String[] { block.toString(), street.toString(), unit.toString(), postalCode.toString() };
```

Comment:
do you notice the code here has a neat arrow head shape? it is generally an indication of SLAP violation. see if you can do better than this here.